service/create/secrets.go

```diff
@@ -0,0 +1,72 @@
+package create
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	microerror "github.com/giantswarm/microkit/error"
+	"k8s.io/client-go/pkg/api"
+	apierrs "k8s.io/client-go/pkg/api/errors"
+	"k8s.io/client-go/pkg/api/v1"
+	"k8s.io/client-go/pkg/watch"
+)
+
+const (
+	componentLabel string        = "clusterComponent"
+	clusterIDLabel string        = "clusterID"
+	secretTimeOut  time.Duration = 10
+)
+
+func (s *Service) getCertsFromSecrets(clusterID string) (map[string]map[string][]byte, error) {
+	components := []string{
+		"apiserver",
+		"worker",
+		"etcd",
+		"calico",
+	}
+	assetTypes := []string{"crt", "key", "ca"}
+	// e.g. certs["worker"]["crt"
+	certs := make(map[string]map[string][]byte)
+
+	watcher, err := s.k8sClient.Core().Secrets(api.NamespaceDefault).Watch(v1.ListOptions{
+		LabelSelector: fmt.Sprintf("%s in (%s), %s=%s", componentLabel, strings.Join(components, ","), clusterIDLabel, clusterID),
+	})
+	if err != nil {
+		return nil, microerror.MaskAny(err)
+	}
+
+	for i := 0; i < len(components); i++ {
```

Comment:
We can simply range over components to make our lives a little bot more easy. See https://play.golang.org/p/co3WmQjILV.